src/coreclr/jit/importer.cpp

```diff
@@ -22857,6 +22857,22 @@ void Compiler::considerGuardedDevirtualization(GenTreeCall*            call,
             return;
         }
 
+        CORINFO_CLASS_HANDLE definingClass = info.compCompHnd->getMethodClass(likelyMethod);
+        likelyClassAttribs = info.compCompHnd->getClassAttribs(definingClass);
+
+        // For instance methods on value classes we need an extended check to
+        // check for the unboxing stub. This is NYI.
+        // Note: Normally likelyMethod above will be the unboxing stub which
+        // would fail GDV for other reasons.
+        // However, with stale profiles or textual PGO input it is still
+        // possible that likelyMethod is not the unboxing stub.
+        // So we do need this explicit check.
+        if ((likelyClassAttribs & CORINFO_FLG_VALUECLASS) != 0)
+        {
+            JITDUMP("Cannot currently handle devirtualizing delegate calls on value types, sorry\n");
```

Comment:
I like how GDV apologizes in different places when it's not able to devirtualize 🙂

Reply:
Best be polite!